### ISSUE TITLE
Crash when opening a page with media for editing

### ISSUE THREAD

**Issue by kean** [wordpress-mobile/WordPress-iOS#21326]
### Steps to reproduce the behavior

- Open a page with media

> [!IMPORTANT]
> It seems to crash only the first time you open a page. You need a fresh install to verify the fix.

##### Tested on [device], iOS [version], Jetpack iOS / WordPress iOS [version]

iPhone 13, iOS 16.6, Jetpack `trunk` (Aug 14)

<img width="1138" alt="Screenshot 2023-08-14 at 9 01 16 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/73678d0f-910a-4450-b277-c8b6678dbf8b">

```
Could not cast value of type 'WordPress.Page' (0x1606893d0) to 'WordPress.Post' (0x105d302a0).
2023-08-14 09:23:38.488059-0400 Jetpack[36581:12011785] Could not cast value of type 'WordPress.Page' (0x1606893d0) to 'WordPress.Post' (0x105d302a0).
```

Sentry issue in 23.0.0.2: https://a8c.sentry.io/issues/4393139751

**Comment by kean** [comment]
It was a regression introduced in the latest build.

**Comment by guarani** [comment]
👋 @kean, there's a Sentry issue that resembles this issue with a couple of events in 23.0: [JETPACK-IOS-11DK](https://a8c.sentry.io/share/issue/adce310ff0044c068cf39ef49140664b/). Do you know if it's the same or a separate issue? 3 of the 4 events are yours and one is from another user.

**Comment by guarani** [comment]
> [JETPACK-IOS-11DK](https://a8c.sentry.io/share/issue/adce310ff0044c068cf39ef49140664b/)

I assigned this Sentry issue to you @kean (as per the crash triage guidelines I'm assigning it to a developer) so that others can see on Sentry that this is being looked into to, but please let me know if I should re-assign it 👍

**Comment by kean** [comment]
I found the issue in one of the 23.0 TestFlight builds and fixed it in 23.0.0.3. There are no similar crashes in 23.0.0.4, which is the production build.

**Comment by guarani** [comment]
Good point about the crashes not being in the final production build. I've marked the Sentry issue as resolved.
